File: javascript/anagram/anagram.js
export class Anagram {
  constructor(word) {
    this.word = word;
    this.sorted = word.toLowerCase().split('').sort().join('');
  }

  matches(potentialAnagrams) {
    return potentialAnagrams.filter(w => w.toLowerCase() !== this.word.toLowerCase() && w.toLowerCase().split('').sort().join('') === this.sorted);
  }
}

File: python/anagram/anagram.py
def find_anagrams(word, candidates):
    alphabetically = ''.join(sorted(word.lower()))
    anagrams = []
    for candidate in candidates:
        if alphabetically == ''.join(sorted(candidate.lower())) and word.lower() != candidate.lower():
            anagrams.append(candidate)
    return anagrams